import os
import sys
import inspect
currentdir = os.path.dirname(os.path.abspath(inspect.getfile(inspect.currentframe())))
parentdir = os.path.dirname(currentdir)
sys.path.insert(0, parentdir)


from PyQt5 import QtCore, QtGui, QtWidgets


class Ui_Admin(object):
    def setupUi(self, Admin):
        Admin.setObjectName("Admin")
        Admin.resize(846, 323)
        self.centralwidget = QtWidgets.QWidget(Admin)
        self.centralwidget.setObjectName("centralwidget")
        self.buttonDeconectare = QtWidgets.QPushButton(self.centralwidget)
        self.buttonDeconectare.setGeometry(QtCore.QRect(690, 120, 141, 81))
        self.buttonDeconectare.setObjectName("buttonDeconectare")
        self.buttonCreate = QtWidgets.QPushButton(self.centralwidget)
        self.buttonCreate.setGeometry(QtCore.QRect(0, 0, 151, 71))
        self.buttonCreate.setObjectName("buttonCreate")
        self.textUpdateOldParola = QtWidgets.QTextEdit(self.centralwidget)
        self.textUpdateOldParola.setGeometry(QtCore.QRect(360, 170, 104, 31))
        self.textUpdateOldParola.setObjectName("textUpdateOldParola")
        self.textUpdateOldCont = QtWidgets.QTextEdit(self.centralwidget)
        self.textUpdateOldCont.setGeometry(QtCore.QRect(560, 170, 104, 31))
        self.textUpdateOldCont.setObjectName("textUpdateOldCont")
        self.textCreateNume = QtWidgets.QTextEdit(self.centralwidget)
        self.textCreateNume.setGeometry(QtCore.QRect(150, 40, 131, 31))
        self.textCreateNume.setObjectName("textCreateNume")
        self.buttonRead = QtWidgets.QPushButton(self.centralwidget)
        self.buttonRead.setGeometry(QtCore.QRect(0, 70, 151, 71))
        self.buttonRead.setObjectName("buttonRead")
        self.label_26 = QtWidgets.QLabel(self.centralwidget)
        self.label_26.setGeometry(QtCore.QRect(560, 140, 101, 31))
        self.label_26.setAlignment(QtCore.Qt.AlignCenter)
        self.label_26.setObjectName("label_26")
        self.textCreateParola = QtWidgets.QTextEdit(self.centralwidget)
        self.textCreateParola.setGeometry(QtCore.QRect(280, 40, 131, 31))
        self.textCreateParola.setObjectName("textCreateParola")
        self.label_30 = QtWidgets.QLabel(self.centralwidget)
        self.label_30.setGeometry(QtCore.QRect(150, 10, 131, 31))
        self.label_30.setAlignment(QtCore.Qt.AlignCenter)
        self.label_30.setObjectName("label_30")
        self.textUpdateOldNume = QtWidgets.QTextEdit(self.centralwidget)
        self.textUpdateOldNume.setGeometry(QtCore.QRect(260, 170, 101, 31))
        self.textUpdateOldNume.setObjectName("textUpdateOldNume")
        self.label_23 = QtWidgets.QLabel(self.centralwidget)
        self.label_23.setGeometry(QtCore.QRect(260, 140, 101, 31))
        self.label_23.setAlignment(QtCore.Qt.AlignCenter)
        self.label_23.setObjectName("label_23")
        self.label_29 = QtWidgets.QLabel(self.centralwidget)
        self.label_29.setGeometry(QtCore.QRect(280, 10, 131, 31))
        self.label_29.setAlignment(QtCore.Qt.AlignCenter)
        self.label_29.setObjectName("label_29")
        self.buttonDelete = QtWidgets.QPushButton(self.centralwidget)
        self.buttonDelete.setGeometry(QtCore.QRect(0, 260, 151, 61))
        self.buttonDelete.setObjectName("buttonDelete")
        self.label_24 = QtWidgets.QLabel(self.centralwidget)
        self.label_24.setGeometry(QtCore.QRect(360, 140, 101, 31))
        self.label_24.setAlignment(QtCore.Qt.AlignCenter)
        self.label_24.setObjectName("label_24")
        self.buttonUpdate = QtWidgets.QPushButton(self.centralwidget)
        self.buttonUpdate.setGeometry(QtCore.QRect(0, 140, 151, 121))
        self.buttonUpdate.setObjectName("buttonUpdate")
        self.label_25 = QtWidgets.QLabel(self.centralwidget)
        self.label_25.setGeometry(QtCore.QRect(460, 140, 101, 31))
        self.label_25.setAlignment(QtCore.Qt.AlignCenter)
        self.label_25.setObjectName("label_25")
        self.label_31 = QtWidgets.QLabel(self.centralwidget)
        self.label_31.setGeometry(QtCore.QRect(530, 10, 131, 31))
        self.label_31.setAlignment(QtCore.Qt.AlignCenter)
        self.label_31.setObjectName("label_31")
        self.label_32 = QtWidgets.QLabel(self.centralwidget)
        self.label_32.setGeometry(QtCore.QRect(410, 10, 121, 31))
        self.label_32.setAlignment(QtCore.Qt.AlignCenter)
        self.label_32.setObjectName("label_32")
        self.textCreateCont = QtWidgets.QTextEdit(self.centralwidget)
        self.textCreateCont.setGeometry(QtCore.QRect(530, 40, 131, 31))
        self.textCreateCont.setObjectName("textCreateCont")
        self.textAfisareReadBibliotecar = QtWidgets.QTextEdit(self.centralwidget)
        self.textAfisareReadBibliotecar.setGeometry(QtCore.QRect(150, 70, 511, 71))
        self.textAfisareReadBibliotecar.setObjectName("textAfisareReadBibliotecar")
        self.label_33 = QtWidgets.QLabel(self.centralwidget)
        self.label_33.setGeometry(QtCore.QRect(560, 200, 101, 31))
        self.label_33.setAlignment(QtCore.Qt.AlignCenter)
        self.label_33.setObjectName("label_33")
        self.textUpdateNewParola = QtWidgets.QTextEdit(self.centralwidget)
        self.textUpdateNewParola.setGeometry(QtCore.QRect(360, 230, 104, 31))
        self.textUpdateNewParola.setObjectName("textUpdateNewParola")
        self.label_34 = QtWidgets.QLabel(self.centralwidget)
        self.label_34.setGeometry(QtCore.QRect(260, 200, 101, 31))
        self.label_34.setAlignment(QtCore.Qt.AlignCenter)
        self.label_34.setObjectName("label_34")
        self.label_35 = QtWidgets.QLabel(self.centralwidget)
        self.label_35.setGeometry(QtCore.QRect(360, 200, 101, 31))
        self.label_35.setAlignment(QtCore.Qt.AlignCenter)
        self.label_35.setObjectName("label_35")
        self.textUpdateNewCont = QtWidgets.QTextEdit(self.centralwidget)
        self.textUpdateNewCont.setGeometry(QtCore.QRect(560, 230, 104, 31))
        self.textUpdateNewCont.setObjectName("textUpdateNewCont")
        self.textUpdateNewNume = QtWidgets.QTextEdit(self.centralwidget)
        self.textUpdateNewNume.setGeometry(QtCore.QRect(260, 230, 101, 31))
        self.textUpdateNewNume.setObjectName("textUpdateNewNume")
        self.label_36 = QtWidgets.QLabel(self.centralwidget)
        self.label_36.setGeometry(QtCore.QRect(460, 200, 101, 31))
        self.label_36.setAlignment(QtCore.Qt.AlignCenter)
        self.label_36.setObjectName("label_36")
        self.label = QtWidgets.QLabel(self.centralwidget)
        self.label.setGeometry(QtCore.QRect(150, 140, 101, 61))
        self.label.setAlignment(QtCore.Qt.AlignCenter)
        self.label.setWordWrap(True)
        self.label.setObjectName("label")
        self.label_2 = QtWidgets.QLabel(self.centralwidget)
        self.label_2.setGeometry(QtCore.QRect(150, 200, 101, 61))
        self.label_2.setAlignment(QtCore.Qt.AlignCenter)
        self.label_2.setWordWrap(True)
        self.label_2.setObjectName("label_2")
        self.textStergereNume = QtWidgets.QTextEdit(self.centralwidget)
        self.textStergereNume.setGeometry(QtCore.QRect(260, 290, 101, 31))
        self.textStergereNume.setObjectName("textStergereNume")
        self.label_37 = QtWidgets.QLabel(self.centralwidget)
        self.label_37.setGeometry(QtCore.QRect(460, 260, 101, 31))
        self.label_37.setAlignment(QtCore.Qt.AlignCenter)
        self.label_37.setObjectName("label_37")
        self.label_38 = QtWidgets.QLabel(self.centralwidget)
        self.label_38.setGeometry(QtCore.QRect(560, 260, 101, 31))
        self.label_38.setAlignment(QtCore.Qt.AlignCenter)
        self.label_38.setObjectName("label_38")
        self.label_39 = QtWidgets.QLabel(self.centralwidget)
        self.label_39.setGeometry(QtCore.QRect(360, 260, 111, 31))
        self.label_39.setAlignment(QtCore.Qt.AlignCenter)
        self.label_39.setObjectName("label_39")
        self.label_40 = QtWidgets.QLabel(self.centralwidget)
        self.label_40.setGeometry(QtCore.QRect(260, 260, 101, 31))
        self.label_40.setAlignment(QtCore.Qt.AlignCenter)
        self.label_40.setObjectName("label_40")
        self.textStergereParola = QtWidgets.QTextEdit(self.centralwidget)
        self.textStergereParola.setGeometry(QtCore.QRect(360, 290, 101, 31))
        self.textStergereParola.setObjectName("textStergereParola")
        self.textStergereCont = QtWidgets.QTextEdit(self.centralwidget)
        self.textStergereCont.setGeometry(QtCore.QRect(560, 290, 111, 31))
        self.textStergereCont.setObjectName("textStergereCont")
        self.textLabelCreateRol = QtWidgets.QLabel(self.centralwidget)
        self.textLabelCreateRol.setGeometry(QtCore.QRect(410, 40, 121, 31))
        self.textLabelCreateRol.setAlignment(QtCore.Qt.AlignCenter)
        self.textLabelCreateRol.setObjectName("textLabelCreateRol")
        self.textLabelUpdateRolOld = QtWidgets.QLabel(self.centralwidget)
        self.textLabelUpdateRolOld.setGeometry(QtCore.QRect(460, 170, 101, 31))
        self.textLabelUpdateRolOld.setAlignment(QtCore.Qt.AlignCenter)
        self.textLabelUpdateRolOld.setObjectName("textLabelUpdateRolOld")
        self.textLabelUpdateRolNew = QtWidgets.QLabel(self.centralwidget)
        self.textLabelUpdateRolNew.setGeometry(QtCore.QRect(460, 230, 101, 31))
        self.textLabelUpdateRolNew.setAlignment(QtCore.Qt.AlignCenter)
        self.textLabelUpdateRolNew.setObjectName("textLabelUpdateRolNew")
        self.textLabelDeleteRol = QtWidgets.QLabel(self.centralwidget)
        self.textLabelDeleteRol.setGeometry(QtCore.QRect(460, 290, 91, 31))
        self.textLabelDeleteRol.setAlignment(QtCore.Qt.AlignCenter)
        self.textLabelDeleteRol.setObjectName("textLabelDeleteRol")
        Admin.setCentralWidget(self.centralwidget)

        self.retranslateUi(Admin)
        QtCore.QMetaObject.connectSlotsByName(Admin)

    def retranslateUi(self, Admin):
        _translate = QtCore.QCoreApplication.translate
        Admin.setWindowTitle(_translate("Admin", "MainWindow"))
        self.buttonDeconectare.setText(_translate("Admin", "Deconectare"))
        self.buttonCreate.setText(_translate("Bibliotecar", "CreateBibliotecar"))
        self.buttonRead.setText(_translate("Bibliotecar", "ReadBibliotecar"))
        self.label_26.setText(_translate("Bibliotecar", "Cont"))
        self.label_30.setText(_translate("Bibliotecar", "Nume"))
        self.label_23.setText(_translate("Bibliotecar", "Nume"))
        self.label_29.setText(_translate("Bibliotecar", "Parola"))
        self.buttonDelete.setText(_translate("Bibliotecar", "DeleteBibliotecar"))
        self.label_24.setText(_translate("Bibliotecar", "Parola"))
        self.buttonUpdate.setText(_translate("Bibliotecar", "UpdateBibliotecar"))
        self.label_25.setText(_translate("Bibliotecar", "Rol"))
        self.label_31.setText(_translate("Bibliotecar", "Cont"))
        self.label_32.setText(_translate("Bibliotecar", "Rol"))
        self.label_33.setText(_translate("Bibliotecar", "Cont"))
        self.label_34.setText(_translate("Bibliotecar", "Nume"))
        self.label_35.setText(_translate("Bibliotecar", "Parola"))
        self.label_36.setText(_translate("Bibliotecar", "Rol"))
        self.label.setText(_translate("Bibliotecar", "Bibliotecar Old"))
        self.label_2.setText(_translate("Bibliotecar", "Bibliotecar New"))
        self.label_37.setText(_translate("Bibliotecar", "Rol"))
        self.label_38.setText(_translate("Bibliotecar", "Cont"))
        self.label_39.setText(_translate("Bibliotecar", "Parola"))
        self.label_40.setText(_translate("Bibliotecar", "Nume"))
        self.textLabelCreateRol.setText(_translate("Bibliotecar", "bibliotecar"))
        self.textLabelUpdateRolOld.setText(_translate("Bibliotecar", "bibliotecar"))
        self.textLabelUpdateRolNew.setText(_translate("Bibliotecar", "bibliotecar"))
        self.textLabelDeleteRol.setText(_translate("Bibliotecar", "bibliotecar"))

        def clickDeconectare():
            from Presenter.adminP import AdminP
            p = AdminP(self)
            p.deconect()
            Admin.close()

        self.buttonDeconectare.clicked.connect(clickDeconectare)

        def clickCreare():
            from Presenter.adminP import AdminP
            p = AdminP(self)
            p.creareBibliotecar()

        self.buttonCreate.clicked.connect(clickCreare)

        def clickVizualizare():
            from Presenter.adminP import AdminP
            p = AdminP(self)
            p.vizualizareBibliotecari()

        self.buttonRead.clicked.connect(clickVizualizare)

        def clickUpdate():
            from Presenter.adminP import AdminP
            p = AdminP(self)
            p.updateBibliotecar()

        self.buttonUpdate.clicked.connect(clickUpdate)

        def clickDelete():
            from Presenter.adminP import AdminP
            p = AdminP(self)
            p.deleteBibliotecar()

        self.buttonDelete.clicked.connect(clickDelete)


if __name__ == "__main__":
    import sys
    app = QtWidgets.QApplication(sys.argv)
    Bibliotecar = QtWidgets.QMainWindow()
    ui = Ui_Admin()
    ui.setupUi(Bibliotecar)
    Bibliotecar.show()
    sys.exit(app.exec_())
